sympy/integrals/integrals.py

```diff
@@ -444,6 +500,20 @@ def doit(self, **hints):
             function = function.doit(**hints)
         if function.is_zero:
             return S.Zero
+        # check for divergent integrals
+        if len(self.limits) == 1:
+            xab = self.limits[0]
+            if len(xab) == 3 and not xab[1:].free_symbols:
+                (x, a, b) = xab
+                xdom = Interval(Min(a,b),Max(a,b))
+                # check if one limit is oo and function limit at that is oo and
+                # check function non positive or non negative in integration limits
+                infl = [i for i in (a,b) if i.is_infinite]
+                if len(infl) > 0 and any([function.limit(x, i).is_infinite for i in infl]):
+                    if solve_univariate_inequality(function>=0, x, domain=xdom, relational=False) == xdom:
+                        return S.Infinity if a < b else S.NegativeInfinity
+                    elif solve_univariate_inequality(function<=0, x, domain=xdom, relational=False) == xdom:
+                        return S.NegativeInfinity if a < b else S.Infinity
```

Comment:
I feel that having to find the limit of the integrand and solve for x is very onerous for most use cases of `integrate`. It adds significant computation that will very rarely be taken advantage of. Most of the time, it pops out `oo` fast enough anyway.

Reply:
So should we make it the last choice in the heuristics for integration. Since this is added to take care of integrals which sympy can't integrate but if they are divergent it can simply give `oo`

Reply:
That would probably be better. Note that divergence does not mean infinite. `Integral(-1, (x, -oo, oo))` is negative infinite and `Integral(sin(x), (x, -oo, oo))` is neither of those.